system/jlib/jdebug.hpp

```diff
@@ -225,6 +225,43 @@ class jlib_decl HiresTimer
     unsigned start;
 };
 
+//---------------------------------------------------------------------------------------------------------------------
+
+/*
+  There are many situations where you want to report or check something periodically.  This class encapsulates th code
+  to perform the check.
+  The constructor has an option to indicate whether the first hasElapsed call should return true, or be suppressed.
+*/
+class PeriodicTimer
+{
+public:
+    PeriodicTimer(unsigned ms, bool suppressFirst)
```

Comment:
suggestion: perhaps add an empty ctor and a reset(unsigned ms, bool suppressFirst),
so instance doesn't need initializing (e.g. if member) immediately , and can be reused if different delay wanted.